clarifai/client/model.py

```diff
@@ -1,22 +1,195 @@
+    Args:
+        filepath (str): The filepath to predict.
+        input_type (str): The type of input. Can be 'image', 'text', 'video' or 'audio.
+
+    Example:
+        >>> from clarifai.client.model import Model
+        >>> model = Model(model_id='model_id', user_id='user_id', app_id='app_id')
+        >>> model_prediction = model.predict_by_filepath('/path/to/image.jpg', 'image')
+        >>> model_prediction = model.predict_by_filepath('/path/to/text.txt', 'text')
+    """
+    if input_type not in ['image', 'text', 'video', 'audio']:
+      raise UserError('Invalid input type it should be image, text, video or audio.')
+    if not os.path.isfile(filepath):
+      raise UserError('Invalid filepath.')
+
+    with open(filepath, "rb") as f:
+      file_bytes = f.read()
+
+    return self.predict_by_bytes(file_bytes, input_type)
+
+  def predict_by_bytes(self, file_bytes: Union[bytes, str], input_type: str):
+    """Predicts the model based on the given bytes.
+    Args:
+        file_bytes (bytes/str): The bytes or raw text string to predict.
+        input_type (str): The type of input. Can be 'image', 'text', 'video' or 'audio.
+    """
+    if input_type not in ['image', 'text', 'video', 'audio']:
+      raise UserError('Invalid input type it should be image, text, video or audio.')
+    # TODO will obtain proto from input class
+    if input_type == "image":
+      input_proto = resources_pb2.Input(
+          data=resources_pb2.Data(image=resources_pb2.Image(base64=file_bytes)))
+    elif input_type == "text":
+      input_proto = resources_pb2.Input(
+          data=resources_pb2.Data(text=resources_pb2.Text(raw=file_bytes)))
+    elif input_type == "video":
+      input_proto = resources_pb2.Input(
+          data=resources_pb2.Data(video=resources_pb2.Video(base64=file_bytes)))
+    elif input_type == "audio":
+      input_proto = resources_pb2.Input(
+          data=resources_pb2.Data(audio=resources_pb2.Audio(base64=file_bytes)))
+
+    return self.predict(inputs=[input_proto])
+
+  def predict_by_url(self, url: str, input_type: str):
+    """Predicts the model based on the given URL.
+    Args:
+        url (str): The URL to predict.
+        input_type (str): The type of input. Can be 'image', 'text', 'video' or 'audio.
+
+    Example:
+        >>> from clarifai.client.model import Model
+        >>> model = Model(model_id='model_id', user_id='user_id', app_id='app_id')
+        >>> model_prediction = model.predict_by_url('url', 'image')
+    """
+    if input_type not in ['image', 'text', 'video', 'audio']:
+      raise UserError('Invalid input type it should be image, text, video or audio.')
+    # TODO will be obtain proto from input class
+    if input_type == "image":
+      input_proto = resources_pb2.Input(
+          data=resources_pb2.Data(image=resources_pb2.Image(url=url)))
+    elif input_type == "text":
+      input_proto = resources_pb2.Input(data=resources_pb2.Data(text=resources_pb2.Text(raw=url)))
```

Comment:
This should be `resources_pb2.Text(url=url)`??

Reply:
Missed it. Will change to url.